src/aliceVision/voctree/SimpleKmeans.hpp

```diff
@@ -118,7 +138,7 @@ struct InitKmeanspp
         // 0 and this sum, then start compute the sum from the first element again
         // until the partial sum is greater than the number drawn: the
         // the previous element is what we are looking for
-        const float perc = (float)rand() / RAND_MAX;
+        const float perc = threadPercs[omp_get_thread_num()];
```

Comment:
suppose `numTrials == 10` and `threadCount == 4`.
In this case `threadPercs` has only 4 random values to choose from so in the end it's useless to run numTrials because there will be only 4 significative results, the other 6 will be computed again by the same thread. (which is against the spirit of the algorithm as we want to explore `numTrials` different solutions)

Maybe it should be
```c++
std::vector<float> threadPercs(numTrials);
```
and then
```c+++
const float perc = threadPercs[j];
```
?

Reply:
Agreed.

Reply:
Fixed now.